src/sql_injection/detect_sql_injection.rs

```diff
@@ -4,25 +4,53 @@ use super::tokenize_query::tokenize_query;
 use crate::diff_in_vec_len;
 
 const SPACE_CHAR: char = ' ';
+pub struct SqlInjectionDetectionResult {
+    pub detected: bool,
+    pub reason: DetectionReason,
+}
```

Comment:
What do you think of something like

```rs
pub enum SqlInjectionDetectionResult {
    Detected,
    NotDetected(DetectionReason),
}
```

That makes it impossible to have `detected: true` with a reason